warp/Network/Wai/Handler/Warp/RequestHeader.hs

```diff
@@ -35,8 +35,10 @@ parseHeaderLines [] = throwIO $ NotEnoughLines []
 parseHeaderLines (firstLine:otherLines) = do
     (method, path', query, httpversion) <- parseRequestLine firstLine
     let path = H.extractPath path'
-        hdr = map parseHeader otherLines
-    return (method, path', path, query, httpversion, hdr)
+        (hdr, hostCount) = foldr parseValidHeader ([], 0) otherLines
+    if hostCount == 1
```

Comment:
Now that the result of fold is being used strictly, perhaps it should be changed into a `foldl'`?